pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -566,39 +567,33 @@ func (r *Replica) applySnapshot(
 			return err
 		}
 	}
+
+	if len(inSnap.externalSSTs)+len(inSnap.sharedSSTs) == 0 && /* simple */
+		inSnap.SSTSize <= snapshotIngestAsWriteThreshold.Get(&st.SV) /* small */ {
+		ingested = false
+	}
+
 	var ingestStats pebble.IngestOperationStats
 	var writeBytes uint64
-	// TODO: separate ingestions for log and statemachine engine. See:
-	//
-	// https://github.com/cockroachdb/cockroach/issues/93251
-	if len(inSnap.externalSSTs) > 0 || len(inSnap.sharedSSTs) > 0 {
+	if ingested {
 		exciseSpan := desc.KeySpan().AsRawSpanWithNoLocals()
 		if ingestStats, err = r.store.TODOEngine().IngestAndExciseFiles(ctx, inSnap.SSTStorageScratch.SSTs(), inSnap.sharedSSTs, inSnap.externalSSTs, exciseSpan); err != nil {
```

Comment:
Can `inSnap.sharedSSTs` and `inSnap.externalSSTs` be non-nil but zero-length? Would that change the behaviour of `IngestAndExciseFiles`? Double-checking because before this change we were passing nils, and now we only check `len=0` and pass the slices as they are (nil or `[]{}`).